src/partner-access/partner-access.service.ts

```diff
@@ -102,4 +96,51 @@ export class PartnerAccessService {
       .leftJoinAndSelect('partnerAccess.partner', 'partner')
       .getMany();
   }
+
+  async assignPartnerAccess(
+    { user, partnerAccesses }: GetUserDto,
+    partnerAccessCode: string,
+  ): Promise<PartnerAccessEntity> {
+    let totalSessionRemainingCount = 0;
+    let totalSessionRedemmedCount = 0;
+
+    const partnerAccessCodeDetails = await this.getValidPartnerAccessCode(partnerAccessCode);
+
+    partnerAccesses.map(async (pa) => {
+      if (partnerAccessCodeDetails.partner.id === pa.partner.id && pa.active === true) {
+        pa.active = false;
+        await this.partnerAccessRepository.save(pa);
+      } else {
+        totalSessionRemainingCount = totalSessionRemainingCount + pa.therapySessionsRemaining;
+        totalSessionRedemmedCount = totalSessionRedemmedCount + pa.therapySessionsRedeemed;
+      }
+    });
+
+    partnerAccessCodeDetails.userId = user.id;
+    partnerAccessCodeDetails.activatedAt = new Date();
+
+    await this.partnerAccessRepository.save(partnerAccessCodeDetails);
+
+    if (!!partnerAccessCodeDetails.featureLiveChat) {
+      const crispResponse = await getCrispPeopleData(user.email);
```

Comment:
would be good to move 125-142 into the `crisp-api.ts` file in its own `updateCrispProfileAccess`